src/main/webapp/app/overview/course-conversations/layout/conversation-selection-sidebar/conversation-sidebar-section/conversation-sidebar-entry/conversation-sidebar-entry.component.ts

```diff
@@ -103,29 +115,46 @@ export class ConversationSidebarEntryComponent implements OnInit, OnDestroy {
                 takeUntil(this.ngUnsubscribe),
             )
             .subscribe(() => {
-                this.settingsChanged.emit();
+                this.settingsDidChange.emit();
             });
     }
+
     ngOnInit(): void {
-        this.hide$.pipe(debounceTime(100), distinctUntilChanged(), takeUntil(this.ngUnsubscribe)).subscribe((shouldHide) => {
-            this.conversationService.changeHiddenStatus(this.course.id!, this.conversation.id!, shouldHide).subscribe({
+        this.favorite$.pipe(debounceTime(100), distinctUntilChanged(), takeUntil(this.ngUnsubscribe)).subscribe((isFavorite) => {
+            this.conversationService.updateIsFavorite(this.course.id!, this.conversation.id!, isFavorite).subscribe({
```

Comment:
Please do not use the non-null assertion operator (the exclamation mark). Because by doing so you're telling the compiler that this expression can't be null or undefined and the compiler should stop complaining about it! But in case it is undefined (and it can since it is an optional parameter) for whatever reason, this would cause a lot of trouble.

Reply:
I will check `=== undefined` then